### ISSUE TITLE
/retry command causes index out of range error in story generator

### ISSUE THREAD

**Issue by dyeo** [thadunge2/AIDungeon#34]
This issue was recreated on the colab on the newest version (commit `9e3ab21`)

1. Start the game.
2. Start a new game.
3. Start a non-random, custom story, with both a story context and starting story prompt.
4. Progress the story by entering a prompt in.
5. Enter the `/retry` command.

Upon using the `/retry` command, the game will crash and produce the following error:

```python
Traceback (most recent call last):
  File "play.py", line 608, in <module>
    play_aidungeon_2()
  File "play.py", line 461, in play_aidungeon_2
    story_manager.act_with_timeout(last_action)
  File "/content/gdrive/My Drive/AIDungeon/story/story_manager.py", line 208, in act_with_timeout
    return func_timeout(self.inference_timeout, self.act, (action_choice,))
  File "/usr/local/lib/python3.6/dist-packages/func_timeout/dafunc.py", line 108, in func_timeout
    raise_exception(exception)
  File "/usr/local/lib/python3.6/dist-packages/func_timeout/py3_raise.py", line 7, in raise_exception
    raise exception[0] from None
  File "/content/gdrive/My Drive/AIDungeon/story/story_manager.py", line 203, in act
    result = self.generate_result(action_choice)
  File "/content/gdrive/My Drive/AIDungeon/story/story_manager.py", line 211, in generate_result
    block = self.generator.generate(self.story_context() + action)
  File "/content/gdrive/My Drive/AIDungeon/story/story_manager.py", line 198, in story_context
    return self.story.latest_result()
  File "/content/gdrive/My Drive/AIDungeon/story/story_manager.py", line 78, in latest_result
    latest_result += self.actions[-mem_ind] + self.results[-mem_ind]
IndexError: list index out of range
```

**Comment by dyeo** [comment]
Interestingly, `/revert` works fine.

**Comment by thadunge2** [comment]
As I mentioned in the thread, I suspect that this is because it's deleting the old action and then trying to pull the deleted action out to feed it into the generator as part of the prompt. I can only assume a recently merged PR is the cause.

**Comment by ShnitzelKiller** [comment]
sorry lol

**Comment by thadunge2** [comment]
Fixed by #35
